conformance/tests/gateway-http-listener-isolation-with-hostname-intersection.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: http-listener-isolation-with-hostname-intersection
+  namespace: gateway-conformance-infra
+spec:
+  gatewayClassName: "{GATEWAY_CLASS_NAME}"
+  listeners:
+  - name: empty-hostname
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+  - name: wildcard-example-com
+    port: 80
+    protocol: HTTP
+    hostname: "*.example.com"
+    allowedRoutes:
+      namespaces:
+        from: All
+  - name: wildcard-foo-example-com
+    port: 80
+    protocol: HTTP
+    hostname: "*.foo.example.com"
+    allowedRoutes:
+      namespaces:
+        from: All
+  - name: abc-foo-example-com
+    port: 80
+    protocol: HTTP
+    hostname: "abc.foo.example.com"
+    allowedRoutes:
+      namespaces:
+        from: All
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: attaches-to-empty-hostname-with-hostname-intersection
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: http-listener-isolation-with-hostname-intersection
+    namespace: gateway-conformance-infra
+    sectionName: empty-hostname
+  hostnames:
+  - "bar.com"
+  - "*.example.com" # request matching is prevented by the isolation wildcard-example-com listener
```

Comment:
these are more tied to hostname intersection b/w listener and route, should this be part of this test or another one ?

Reply:
My interpretation is that listener intersection is about one listener and its HTTPRoutes (so it could be tested separately) -- https://github.com/kubernetes-sigs/gateway-api/blob/4b1a28d0c874a3816d344fb0d4e2c04424294d54/apis/v1/httproute_types.go#L71-L73

while listener isolation is about multiple listeners  -- https://github.com/kubernetes-sigs/gateway-api/blob/4b1a28d0c874a3816d344fb0d4e2c04424294d54/apis/v1/gateway_types.go#L173-L179

Or to put it differently, correct implementation of listener intersection is required for this test, but ultimately the tests is about listener isolation.